cli/command_repository_set_parameters_test.go

```diff
@@ -127,14 +123,35 @@ func (s *formatSpecificTestSuite) TestRepositorySetParametersUpgrade(t *testing.
 	// default values
 	require.Contains(t, out, "Max pack length:     20 MiB")
 
-	if s.formatVersion == content.FormatVersion1 {
+	switch s.formatVersion {
+	case content.FormatVersion1:
 		require.Contains(t, out, "Format version:      1")
 		require.Contains(t, out, "Epoch Manager:       disabled")
 		env.RunAndExpectFailure(t, "index", "epoch", "list")
-	} else {
+	case content.FormatVersion2:
 		require.Contains(t, out, "Format version:      2")
 		require.Contains(t, out, "Epoch Manager:       enabled")
 		env.RunAndExpectSuccess(t, "index", "epoch", "list")
+	default:
+		require.Contains(t, out, "Format version:      3")
+		require.Contains(t, out, "Epoch Manager:       enabled")
+		env.RunAndExpectSuccess(t, "index", "epoch", "list")
+	}
+
+	env.Environment["KOPIA_UPGRADE_LOCK_ENABLED"] = "1"
+
+	if s.formatVersion < content.MaxFormatVersion {
+		env.RunAndExpectSuccess(t, "repository", "upgrade",
+			"--upgrade-owner-id", "owner",
+			"--io-drain-timeout", "1s", "--force",
+			"--status-poll-interval", "1s",
+			"--max-clock-drift", "1s")
+	} else {
+		env.RunAndExpectFailure(t, "repository", "upgrade",
+			"--upgrade-owner-id", "owner",
+			"--io-drain-timeout", "1s", "--force",
+			"--status-poll-interval", "1s",
+			"--max-clock-drift", "1s")
```

Comment:
What's the difference between the `if` and `else` blocks?

Reply:
`RunAndExpectFailure` vs `RunAndExpectSuccess`

But I can still make this a bit nicer.

Reply:
refactored the code slightly

https://github.com/kopia/kopia/pull/1818/commits/4f4917ec02093d3d222205ad5b3348477f519ca6